Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -1345,6 +1389,34 @@ - (void)setWebPushActionHandler:(WKWebsiteDataStore *(^)(_WKWebPushAction *))han
     RetainPtr unhandled = adoptNS([[NSMutableSet alloc] init]);
 
     for (BSAction *action in applicationActions) {
+        if (action.UIActionType == UIActionTypeNotificationResponse) {
+            // Try to pull out a notification response from the action
+            UNNotificationResponse *response = nil;
+            SEL responseSelector = NSSelectorFromString(@"response");
```

Comment:
Should this be defined in a category method instead so you don't have to `performSelector:`?

Reply:
Hmmmm okay.